tabpfn/scripts/transformer_prediction_interface.py

```diff
@@ -167,8 +173,15 @@ def _validate_targets(self, y):
         return np.asarray(y, dtype=np.float64, order="C")
 
     def fit(self, X, y, overwrite_warning=False):
-        # Check that X and y have correct shape
-        X, y = check_X_y(X, y, force_all_finite=False)
+        """
+        Validates the training set and stores it.
+
+        If no_grad is set to false the function takes X as tensors and doesn't fully check the compatibility.
```

Comment:
This is hard to understand.
Maybe write something like this

If clf.no_grad:
    X,y should be of type np.array
else:
    X, y should both be of type torch.Tensors